### PR TITLE
[DDW-458] Improve Acceptance Tests Coverage

### PR DESCRIPTION
**Pull request by MarcusHurney**
This PR expands the coverage of the Daedalus suite of acceptance tests.

## Todo:

- [x] Add test for `"Connecting to network..."` message and `"Having trouble connecting to network?"` notification in `Loading.js` component.
- [x] Add test for `"Having trouble syncing?"` notification in `Loading.js` component.
- [x] Add test for block synchronization stalling scenario, only if both local and network block height remains the same for 120 seconds, which should result in the `"Network connection lost - reconnecting..."` notification in `Loading.js` component.
- [x] Add test for the Block Consolidation Screen.
- [x] Fix mnemonic word selector in tests using the seed words.

### Basics

- [x] PR is updated to the most recent version of target branch (and there are no conflicts)
- [x] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [x] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [x] Automated tests: All acceptance tests are passing (`yarn run test`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [x] There are no *flow* errors or warnings (`yarn run flow:test`)
- [x] There are no *lint* errors or warnings (`yarn run lint`)
- [x] Text changes are proofread and approved (Jane Wild)
- [x] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [x] UI changes look good in all themes (Alexander Rukin)
- [x] Storybook works and no stories are broken (`yarn run storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [x] Important parts of the code are properly documented and commented
- [x] Code is properly typed with flow
- [x] React components are split-up enough to avoid unnecessary re-rendering
- [x] Any code that only works in Electron is neatly separated from components

### Testing
- [x] New feature / change is covered by acceptance tests
- [x] All existing acceptance tests are still up-to-date
- [x] New feature / change is covered by Daedalus Testing scenario
- [x] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: source/renderer/app/config/timingConfig.js [hunk 1]
```diff
@@ -3,15 +3,15 @@ const { isTest } = global.environment;
 
 // All configuration values for timeouts / intervals should go here
 /* eslint-disable max-len */
-export const REPORT_ISSUE_TIME_TRIGGER = isTest ? 8 : 5 * 60; // 5 minutes / 10 seconds (isTest = true) | unit: seconds
+export const REPORT_ISSUE_TIME_TRIGGER = isTest ? 12 : 5 * 60; // 5 minutes / 12 seconds (isTest = true) | unit: seconds
```
Comment (MarcusHurney) [inline source/renderer/app/config/timingConfig.js]: @nikolaglumac @daniloprates 
When `isTest=true`, the `REPORT_ISSUE_TIME_TRIGGER` and `MAX_ALLOWED_STALL_DURATION` are set to 12 seconds and 15 seconds respectively. This ensures the tests for the loading screen don't need to run for 5 minutes in order to trigger the report issue notification.

Comment (nikolaglumac) [reply source/renderer/app/config/timingConfig.js]: @MarcusHurney can we reduce both timeouts to just a couple of seconds for test environment?

Comment (MarcusHurney) [reply source/renderer/app/config/timingConfig.js]: @nikolaglumac
Yes, that is possible, but it will cause the trouble connecting notification to flash on other acceptance tests that show the loading screen for more than a few seconds. It didn’t seem to be an issue when I ran the full suite of acceptance tests but the values I have set here are exactly the length of time needed for it not to flash on other tests. I am fine with making them shorter though.

Comment (nikolaglumac) [reply source/renderer/app/config/timingConfig.js]: @MarcusHurney I have another idea - let's keep the default value for all other tests and then just set some minimum value like 1 or 2 seconds just for those new tests which actually test those notifications?

Comment (MarcusHurney) [reply source/renderer/app/config/timingConfig.js]: I’ve been thinking about how to distinguish the two tests that use these time constants from the standard `isTest` environment variable that we typically use to differentiate test mode from develop etc, but I am not sure what the best approach is for determining that distinction. Do you know if there is a way to set a flag on `process.env` before each those tests run that I could then add to the `environment` variable and check for in `timingConfig.js`? I am open to other suggestions as well, that’s just one idea.

Comment (nikolaglumac) [reply source/renderer/app/config/timingConfig.js]: @MarcusHurney I was thinking about the approach where we would set the 1 sec values as a part of a test "before" step. That way we could alter these timers only for these new tests.
Not sure if we can do that... Can you try it out?

Comment (MarcusHurney) [reply source/renderer/app/config/timingConfig.js]: I can try that.

**Comment by nikolaglumac** [comment]
@MarcusHurney can you please sort out merge conflicts? 🙏

File: features/step_definitions/block-consolidation-page-steps.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import { When, Then } from 'cucumber';
+import { expect } from 'chai';
+
+const BLOCK_CONSOLIDATION_COMPONENT = '.BlockConsolidationStatus_component';
+const BLOCK_CONSOLIDATION_EXPLANATION = '.BlockConsolidationStatus_content p:nth-child(3)';
+const EPOCHS_CONSOLIDATION_STATUS = '.BlockConsolidationStatus_epochs p span';
+const EPOCHS_CONSOLIDATED = '.BlockConsolidationStatus_indicatorEpochsConsolidated p';
+const TRAILING_BY_2_EPOCH = '.BlockConsolidationStatus_indicatorEpochsBehind p';
+const MAXIMUM_EPOCH = '.BlockConsolidationStatus_fullEpoch';
+const SYNC_PROGRESS = '.BlockConsolidationStatus_indicatorEpochsSynced p span';
```
Comment (DominikGuzei) [inline features/step_definitions/block-consolidation-page-steps.js]: Please wrap those constants into a single object to make it clearer that all of these constants are selectors eg:
```
const SELECTORS = {
  BLOCK_CONSOLIDATION_COMPONENT: '.BlockConsolidationStatus_component',
  …
}
```

which can then be used like `SELECTORS.BLOCK_CONSOLIDATION_COMPONENT`

File: features/step_definitions/ada-redemption-steps.js [hunk 3]
```diff
@@ -221,9 +221,9 @@ When(
         '.AdaRedemptionForm_component .pass-phrase input',
         word
       );
-      await this.client.waitForVisible(`//li[contains(text(), '${word}')]`);
-      await this.waitAndClick(`//li[contains(text(), '${word}')]`);
-      await this.client.waitForVisible(`//span[contains(text(), '${word}')]`);
+      await this.client.waitForVisible(`//li[text()="${word}"]`);
+      await this.waitAndClick(`//li[text()="${word}"]`);
```
Comment (DominikGuzei) [inline features/step_definitions/ada-redemption-steps.js]: `this.waitAndClick` already does `waitForVisible` so you can remove those extra checks

File: features/step_definitions/block-consolidation-page-steps.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+});
+
+Then(/^the Block Consolidation Status Page is (hidden|visible)/, async function (state) {
+  const isVisible = state === 'visible';
+  return await this.client.waitForVisible(BLOCK_CONSOLIDATION_COMPONENT, null, !isVisible);
+});
+
+Then(/^the page accurately renders an explanation of how block consolidation works in file storage$/, async function () {
+  await this.client.waitForText(BLOCK_CONSOLIDATION_EXPLANATION);
+  const explanationText = await this.client.getText(BLOCK_CONSOLIDATION_EXPLANATION);
```
Comment (DominikGuzei) [inline features/step_definitions/block-consolidation-page-steps.js]: You can simplify the those two lines by using the helper `this.getVisibleTextsForSelector(this.client, BLOCK_CONSOLIDATION_EXPLANATION)` (same for all other occasions of `waitForText` + `getText`)

File: features/step_definitions/trouble-connecting-notification-steps.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import { Then, When } from 'cucumber';
+import { waitUntilTextInSelector } from '../support/helpers/shared-helpers';
+
+const LOADING_COMPONENT = '.Loading_component';
+const REPORT_ISSUE_TEXT_H1 = '.Loading_reportIssueText';
+const REPORT_ISSUE_BUTTON = '.Loading_reportIssueButton';
```
Comment (DominikGuzei) [inline features/step_definitions/trouble-connecting-notification-steps.js]: Same here (bundle them into one `SELECTOR` object)

**Comment by MarcusHurney** [comment]
@DominikGuzei
I've implemented the changes you suggested, thanks for those!

@nikolaglumac
I wasn't able to set the trigger issue time constants to 2 seconds for the 2 particular tests here that cover the loading screen. Not sure if it's worth investing any more time into that one considering I burned half a day on it and walked away with nothing 😭

**Comment by nikolaglumac** [comment]
Everything looks good now @MarcusHurney - thanks 🎉
